docs/workflow/building/coreclr/cross-building.md

```diff
@@ -1,184 +1,163 @@
 
--or-
+Once you have all the required dependencies, it is a straightforward process. Windows knows how to cross-build behind curtains, so all you have to do is specify which architecture you want to build for:
 
-    C:\runtime> src\coreclr\build-runtime.cmd -arm -debug
-
-Cross Compilation for ARM64 on macOS
-================================================
-
-The toolset required to do native compilation described in [macOS requirements](https://github.com/dotnet/runtime/blob/main/docs/workflow/requirements/macos-requirements.md) has cross compilation capabilities with no additional installs required. The only needed change is to use the `-cross` flag in the build scripts. For example, a release CoreCLR runtime that targets ARM64 can be built using the following commandline from the root of the repo:
-
-    ./build.sh -subset clr.runtime -c release -cross -arch arm64 /p:CrossBuild=true
-
-Cross Compilation for ARM, ARM64 or x86 on Linux
-================================================
-
-Through cross compilation, on Linux it is possible to build CoreCLR for arm or arm64. Note that this documentation exists to explain using `runtime/eng/common/build-rootfs.sh`. This will build a rootfs and then use it to cross build. Newer documentation [linux-instructions.md](linux-instructions.md) exists which leverages docker to use a prebuilt environment to cross build.
-
-Requirements for targeting Debian based distros
-------------------------------------------------
-
-You need a Debian based host and the following packages need to be installed:
-
-    ~/runtime/ $ sudo apt-get install qemu qemu-user-static binfmt-support debootstrap
+```cmd
+.\build.cmd -s clr -c Release -arch arm64
+```
 
-In addition, to cross compile CoreCLR the binutils for the target are required. So for arm you need:
+### Cross-Compiling for x86 on Windows
 
-    ~/runtime/ $ sudo apt-get install binutils-arm-linux-gnueabihf
+Building for x86 doesn't require any additional software installed or configured, since all the x64 build tools also have the capability of building for x86. Just specify it when calling the build script:
 
-and conversely for arm64:
+```cmd
+.\build.cmd -s clr -c Release -arch x86
+```
 
-    ~/runtime/ $ sudo apt-get install binutils-aarch64-linux-gnu
+## MacOS Cross-Building
 
-and for armel (ARM softfp):
+This section will go over cross-compiling on MacOS. Currently, MacOS allows you to cross-compile between x64 and ARM64.
 
-    ~/runtime/ $ sudo apt-get install binutils-arm-linux-gnueabi
+Similarly to targeting Windows x86, the native tooling you installed back in the [MacOS requirements doc](/docs/workflow/requirements/macos-requirements.md) has the capabilities to effectuate the cross-compilation. You have simply to pass the `-cross` flag, along with the designated architecture. For example, for an arm64 build on an Intel x64 Mac:
 
+```bash
+./build.sh -s clr -c Release --cross -a arm64
```

Comment:
@janvorli - can you please explain to me what are the exact rules when <code>--cross</code> needs to be specified? On Windows, it seems to me that I can easily cross-build for arm / arm64 just by specifying the appropriate targeting architecture.

Reply:
On Unix, you need to specify it always when building for a different architecture than the one the build is running on.

Reply:
Hmm, is that a real requirement or just a deficiency of the related scripts? I mean, I guess we know what the hosting ISA is and we have been given the targeting ISA in the <code>-a</code> argument, why do we need the extra <code>--cross</code> option?

Reply:
`--cross` needs to be specified whenever you are building against a ROOTFS instead of your system libraries. We build with this setting in the following scenarios:

- Target architecture differs from host architecture (as @janvorli said)
- Target libc differs from host libc
  - Example: Building on Ubuntu (glibc) for Alpine (MUSL libc)
  - Example: Building on CBL-Mariner (newer glibc) for RHEL-8 compatibility (older glibc)
- Target distro is different enough from the host distro that we can't build against the host toolset (building on Ubuntu for FreeBSD)

The `--cross` argument is used in determining the RID for the build and for turning off some validation steps that run the tools we built on the machine we built on.

Reply:
Great, thanks Jeremy for clarifying, non-blocking for this particular PR but we might want to have this explained in some publicly available space.

Reply:
Noted. I think indeed our docs need more work, and I'm certainly willing to continue helping. But I think it should be done as a currently ongoing effort (maybe with a .NET 8 milestone?), rather than extensions to a QW work item.